File: TransCoder_tokenized_test_set_functions/python/COUNT_WAYS_INCREASE_LCS_LENGTH_TWO_STRINGS_ONE.py
def waysToIncreaseLCSBy1 ( str1, str2 ) :
    m = len ( str1 )
    n = len ( str2 )
    position = [ [ ] for i in range ( M ) ]
    for i in range ( 1, n + 1, 1 ) :
        position [ ord ( str2 [ i - 1 ] ) - 97 ] . append ( i )
    lcsl = [ [ 0 for i in range ( n + 2 ) ] for j in range ( m + 2 ) ]
    lcsr = [ [ 0 for i in range ( n + 2 ) ] for j in range ( m + 2 ) ]
    for i in range ( 1, m + 1, 1 ) :
        for j in range ( 1, n + 1, 1 ) :
            if ( str1 [ i - 1 ] == str2 [ j - 1 ] ) :
                lcsl [ i ] [ j ] = 1 + lcsl [ i - 1 ] [ j - 1 ]
            else :
                lcsl [ i ] [ j ] = max ( lcsl [ i - 1 ] [ j ], lcsl [ i ] [ j - 1 ] )
    for i in range ( m, 0, - 1 ) :
        for j in range ( n, 0, - 1 ) :
            if ( str1 [ i - 1 ] == str2 [ j - 1 ] ) :
                lcsr [ i ] [ j ] = 1 + lcsr [ i + 1 ] [ j + 1 ]
            else :
                lcsr [ i ] [ j ] = max ( lcsr [ i + 1 ] [ j ], lcsr [ i ] [ j + 1 ] )
    ways = 0
    for i in range ( 0, m + 1, 1 ) :
        for C in range ( 0, 26, 1 ) :
            for j in range ( 0, len ( position [ C ] ), 1 ) :
                p = position [ C ] [ j ]
                if ( lcsl [ i ] [ p - 1 ] + lcsr [ i + 1 ] [ p + 1 ] == lcsl [ m ] [ n ] ) :
                    ways += 1
    return ways

File: TransCoder_tokenized_test_set_functions/java/COUNT_WAYS_INCREASE_LCS_LENGTH_TWO_STRINGS_ONE.java
static int waysToIncreaseLCSBy1 ( String str1, String str2 ) {
  int m = str1 . length ( ), n = str2 . length ( );
  Vector < Integer > [ ] position = new Vector [ M ];
  for ( int i = 0;
  i < M;
  i ++ ) position [ i ] = new Vector < > ( );
  for ( int i = 1;
  i <= n;
  i ++ ) position [ str2 . charAt ( i - 1 ) - 'a' ] . add ( i );
  int [ ] [ ] lcsl = new int [ m + 2 ] [ n + 2 ];
  int [ ] [ ] lcsr = new int [ m + 2 ] [ n + 2 ];
  for ( int i = 0;
  i <= m + 1;
  i ++ ) for ( int j = 0;
  j <= n + 1;
  j ++ ) lcsl [ i ] [ j ] = lcsr [ i ] [ j ] = 0;
  for ( int i = 1;
  i <= m;
  i ++ ) {
    for ( int j = 1;
    j <= n;
    j ++ ) {
      if ( str1 . charAt ( i - 1 ) == str2 . charAt ( j - 1 ) ) lcsl [ i ] [ j ] = 1 + lcsl [ i - 1 ] [ j - 1 ];
      else lcsl [ i ] [ j ] = Math . max ( lcsl [ i - 1 ] [ j ], lcsl [ i ] [ j - 1 ] );
    }
  }
  for ( int i = m;
  i >= 1;
  i -- ) {
    for ( int j = n;
    j >= 1;
    j -- ) {
      if ( str1 . charAt ( i - 1 ) == str2 . charAt ( j - 1 ) ) lcsr [ i ] [ j ] = 1 + lcsr [ i + 1 ] [ j + 1 ];
      else lcsr [ i ] [ j ] = Math . max ( lcsr [ i + 1 ] [ j ], lcsr [ i ] [ j + 1 ] );
    }
  }
  int ways = 0;
  for ( int i = 0;
  i <= m;
  i ++ ) {
    for ( char d = 0;
    d < 26;
    d ++ ) {
      for ( int j = 0;
      j < position [ d ] . size ( );
      j ++ ) {
        int p = position [ d ] . elementAt ( j );
        if ( lcsl [ i ] [ p - 1 ] + lcsr [ i + 1 ] [ p + 1 ] == lcsl [ m ] [ n ] ) ways ++;
      }
    }
  }
  return ways;
}